com_migratetojoomla/admin/src/Helper/FtpHelper.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_migratetojoomla
+ *
+ * @copyright   (C) 2006 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Component\MigrateToJoomla\Administrator\Helper;
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Client\FtpClient;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+class FtpHelper
+{
+
+    public static $options = array();
+    public static $ftp = false;
+    /**
+     * Method to check Enter base url connection
+     * 
+     * @param string Base url of live website
+     * @return boolean True on success
+     * 
+     * @since 1.0
+     */
+    public static function testConnection($data = [])
+    {
+        FtpHelper::$options['host'] = $data['ftphost'];
+        FtpHelper::$options['port'] = $data['ftpport'];
+        FtpHelper::$options['username'] = $data['ftpusername'];
+        FtpHelper::$options['password'] = $data['ftppassword'];
+        FtpHelper::$options['basedir'] = $data['ftpbasefir'];
+
+        $response = FtpHelper::login();
+
+        $app = Factory::getApplication();
+
+        if ($response) {
+            $app->enqueueMessage("Ftp connection Success", 'success');
+        }else{
+            $app->enqueueMessage("Ftp connection Unsuccessful", 'danger');
+        }
+    }
+
+    /**
+     * Method to list files in a directory
+     * 
+     * @param string Directory
+     * @return array List of files
+     * 
+     * @since  1.0
+     */
+    public static function listDirectory($directory)
+    {
+    }
```

Comment:
If you have several classes that all have the same functions, you could create an interface that would set the definition for all the functions you need and have FtpHelper, HttpHelper... implement that interface.
It is just a suggestion.